### PR TITLE
feat: Implements the table and column lineage for Neo4j 

### PR DESCRIPTION
**Pull request by verdan**
### Summary of Changes

This PR implements the Neo4j lineage backend for Table and Columns both. 

### Tests

Adds the new test cases for this new implementation. 

### Documentation

NA

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

**Comment by verdan** [comment]
@sewardgw would you mind verifying this with your databuilder data?

File: metadata_service/proxy/neo4j_proxy.py [hunk 1]
```diff
@@ -1408,7 +1408,124 @@ def get_resources_using_table(self, *,
...
+        Retrieves the lineage information for the specified resource type.
+
+        :param id: key of a table or a column
+        :param resource_type: Type of the entity for which lineage is being retrieved
+        :param direction: Whether to get the upstream/downstream or both directions
+        :param depth: depth or level of lineage information
+        :return: The Lineage object with upstream & downstream lineage items
+        """
+
+        def _get_badges(badges: list) -> List[Badge]:
```
Comment (sewardgw) [inline metadata_service/proxy/neo4j_proxy.py:1424]: Just curious if there is somewhere else that this is being used or could be used? This seems like a reusable function across tables & lineage

Comment (dorianj) [reply metadata_service/proxy/neo4j_proxy.py:1424]: unrelated to other comment: how does `list` typecheck? Isn't `List` required?

Comment (verdan) [reply metadata_service/proxy/neo4j_proxy.py]: `List` is suitable for annotating return types. For arguments and other cases builtin `list` is fine. (or Iterable).

Comment (verdan) [reply metadata_service/proxy/neo4j_proxy.py]: > Just curious if there is somewhere else that this is being used or could be used? This seems like a reusable function across tables & lineage

@sewardgw good catch, refactored.

**Review by dorianj — APPROVED** [review]

**Pull request merged**